src/render.tsx

```diff
@@ -85,6 +90,10 @@ export async function render<T>(options: AfterRenderProps<T>) {
     match: reactRouterMatch,
   });
 
+  if(!html || html.length === 0) {
```

Comment:
not sure i understand this one.

Reply:
if not html is rendered/returned from the `renderToString`, will return a 404.

Reply:
Further to this, if we do not send this explicit 404 to the client, anything inside Document will be sent to the client.

Reply:
Hi @jtart - does this solve your use case?

Right now, you can define a catch-all path in your `routes` like:
```
  // add this as the last route
  {
    path: '/', // you need to define path for getInitialProps to work in component
    component: NotFound,
    exact: false,
  },
```
and the NotFound component can look like this:
```
class NotFound extends React.Component {
  static async getInitialProps(ctx: any) {
    if (ctx.res) {
      ctx.res.statusCode = 404;
    }
  }

  render() {
    return (
      <div>
        <Header />
        <h1>404 Not Found</h1>
        <p>Page not found</p>
        <Footer />
      </div>
    );
  }
}
```
IMHO nothing more is needed.